File: src/main/java/UI/MainWindow.java
package UI;

import UI.Scene.SceneManager;
import java.awt.*;
import javax.swing.*;

public class MainWindow {
    private JFrame mainFrame;

    public MainWindow() {
        render();

        SceneManager sceneManager = new SceneManager(mainFrame);
        sceneManager.transitionToMainMenu();

        // timer.scheduleAtFixedRate(() -> timePassed(1), 1, 1, TimeUnit.MINUTES);

        // TODO this has to happen in game scene only

    }

    private void render() {
        mainFrame = new JFrame("Tamagotchi game");
        mainFrame.setSize(900, 600);
        mainFrame.setLayout(new GridLayout());
        mainFrame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        mainFrame.setLocationRelativeTo(null);

        mainFrame.setVisible(false);
    }
}

File: src/main/kotlin/UI/MainWindow.kt
package UI

import UI.Scene.SceneManager
import java.awt.GridLayout
import javax.swing.JFrame
import javax.swing.WindowConstants

class MainWindow {
    private var mainFrame: JFrame = JFrame("Tamagotchi game")

    init {
        render()

        val sceneManager = SceneManager(mainFrame)
        sceneManager.transitionToMainMenu()

        // timer.scheduleAtFixedRate({ timePassed(1) }, 1, 1, TimeUnit.MINUTES);

        // TODO this has to happen in game scene only
    }

    private fun render() {
        mainFrame.setSize(900, 600)
        mainFrame.layout = GridLayout()
        mainFrame.defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE
        mainFrame.setLocationRelativeTo(null)
        mainFrame.isVisible = false
    }
}